packages/dbgpt-core/src/dbgpt/observability/default_provider.py

```
@@ -0,0 +1,482 @@
+            rows = q.group_by(bucket).order_by(bucket).all()
+            points = []
+            for r in rows:
+                points.append(
+                    TimeseriesPoint(
+                        timestamp=datetime.strptime(r.bucket, py_fmt),
+                        value=float(r.value) if r.value is not None else 0.0,
+                    )
+                )
+            return Timeseries(metric=metric, points=points)
+
+    def _apply_trace_filter(self, q, filter: TraceFilter):
```

Ruff (0.16.2)

[error] Function argument `filter` is shadowing a Python builtin

(A002)